test/conftest.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python3
+# coding=utf-8
+"""
+Variables used across test modules, including the setup and teardown fixture
+that is run before and after every test
+"""
+import os
+import shutil
+import pytest
+
+# sys.path.append("..")
+from ad_logger import ad_logger
+from toolbox import toolbox
+from config import ad_config
+
+# Variables used across test classes
+
+# TODO prevent logging writing to syslog when in testing mode
+
+data_dir = os.path.abspath("test/data/")
+# Temporary directories to copy test files into and to contain outputs
+tempdir = os.path.join(os.path.abspath("test"), "temp/")
+temp_testfiles_dir = os.path.join(tempdir, "demultiplex_test_files")
+temp_runfolderdir = os.path.join(temp_testfiles_dir, "test_runfolders/")
+# temp_samplesheets_dir = os.path.join(temp_runfolderdir, "samplesheets")
+# temp_samplesheet_path = os.path.join(temp_samplesheets_dir, "%s_SampleSheet.csv")
+temp_log_dir = os.path.join(tempdir, "automate_demultiplexing_logfiles")
+
+# Temp directory for SampleSheet validator SampleSheet test cases
+sv_samplesheet_temp_dir = os.path.join(tempdir, "samplesheets")
+
+
+@pytest.fixture(scope="function")
+def logger_obj():
+    temp_log = os.path.join(tempdir, "temp.log")
+    return ad_logger.AdLogger("demultiplex", "demultiplex", temp_log).get_logger()
+
+
```

Comment:
Is there a reason you didn't use the pytest builtin fixtures tmp_path or tmp_factory_path for creating the temp file structure for testing?

Reply:
Have created github issue